ocaml-lsp-server/test/e2e/__tests__/textDocument-completion.test.ts

```diff
@@ -425,18 +425,201 @@ describe_opt("textDocument/completion", () => {
   });
 
   it("completion doesn't autocomplete record fields", async () => {
-    openDocument(outdent`
+    await openDocument(outdent`
       type r = {
         x: int;
         y: string
       }
 
-      let _ = 
+      let _ =
     `);
 
     let items: Array<any> = await queryCompletion(Types.Position.create(5, 8));
     expect(
       items.filter((compl) => compl.label === "x" || compl.label === "y"),
     ).toHaveLength(0);
   });
+
+  it("completion for holes", async () => {
+    await openDocument(outdent`
+let u : Lexing.position = _
```

Comment:
Could you modify this test to be:

* Less noisy. We don't need so many candidates in the results.
* Rely on the standard library as little as possible. That would mean defining all the types you are using. This would make the test more robust when upgrading to newer versions of OCaml

Reply:
I updated the test to be minimally noisy, but the problem is that along with calling `construct`, we also look for completion for `_` prefix, which brings a lot of results such as `__FILE__`, `__POS__`, which we can't rid of (?), unless we do not offer completion when prefix is `_`. I thought about not offering completion on `_`, since usually variables starting with `_` are "not used", but that seems limiting from a user's perspective, if that makes sense.